### PR TITLE
Avoid adding hooks for globalThis in MODULARIZE mode

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #22588

### CODE REVIEW

File: src/runtime_debug.js [hunk 1]
```diff
@@ -56,6 +56,14 @@ function isExportedByForceFilesystem(name) {
...
  */
 function hookGlobalSymbolAccess(sym, func) {
+#if MODULARIZE && !EXPORT_ES6
+  // In MODULARIZE mode the generated code runs inside a function scope and not
+  // the global scope, and JavaScript does not provide access to function scopes
+  // so we cannot dynamically modify the scrope using `defineProperty` in this
+  // case.
+  //
+  // In this mode we simply ignore requests for `hookGlobalSymbolAccess`.
+#else
```
Comment (kripken) [inline src/runtime_debug.js:66]: To make it clearer why it is valid to ignore here, how about renaming this function to `addWarningOnGlobalSymbol`? By saying that it is a warning, it is clear that it is optional.

Comment (sbc100) [reply src/runtime_debug.js:66]: OK, will do, however all functions in this file are debug only so I think it should fairly clear.

Comment (kripken) [reply src/runtime_debug.js:66]: Ah, good point. In that case lgtm with just a comment here, like "As everything else in this file, these are just debug checks, so we simply avoid them in this case."

**Comment by juj** [comment]
Yeh, this is probably simplest, rather than trying to codegen the checks. Thanks for thinking about this!

**Pull request merged**